clients/fides-js/src/fides-tcf.ts

```diff
@@ -0,0 +1,138 @@
+/**
+ * Fides-tcf.js: Javascript library for Fides (https://github.com/ethyca/fides), including
+ * features for supporting the Transparency Consent Framework
+ *
+ * This JS module provides easy access to interact with Fides from a webpage, including the ability to:
+ * - initialize the page with default consent options (e.g. opt-out of advertising cookies, opt-in to analytics, etc.)
+ * - read/write the current user's consent preferences to their browser as a cookie
+ * - push the current user's consent preferences to other systems via integrations (Google Tag Manager, Meta, etc.)
+ *
+ * See https://fid.es for more information!
+ *
+ * Basic usage of this module in an HTML page is:
+ * ```
+ * <script src="https://privacy.{company}.com/fides.js"></script>
+ * <script>
+ *   window.Fides.init({
+ *     consent: {
+ *       options: [{
+ *         cookieKeys: ["data_sales"],
+ *         default: true,
+ *         fidesDataUseKey: "advertising"
+ *       }]
+ *     },
+ *     experience: {},
+ *     geolocation: {},
+ *     options: {
+ *           debug: true,
+ *           isDisabled: false,
+ *           isGeolocationEnabled: false,
+ *           geolocationApiUrl: "",
+ *           overlayParentId: null,
+ *           modalLinkId: null,
+ *           privacyCenterUrl: "http://localhost:3000"
+ *         }
+ *   });
+ * </script>
+ * ```
+ *
+ * ...and later:
+ * ```
+ * <script>
+ *   // Query user consent preferences
+ *   if (window.Fides.consent.data_sales) {
+ *     // ...enable advertising scripts
+ *   }
+ * ```
+ */
+import { gtm } from "./integrations/gtm";
+import { meta } from "./integrations/meta";
+import { shopify } from "./integrations/shopify";
+
+import { FidesConfig } from "./lib/consent-types";
+
+import { tcf } from "./lib/tcf";
+import {
+  getInitialCookie,
+  getInitialFides,
+  initialize,
+} from "./lib/initialize";
+import type { Fides } from "./lib/initialize";
+import { dispatchFidesEvent } from "./lib/events";
+import { isNewFidesCookie } from "./fides";
+import { renderOverlay } from "./lib/tcf/renderOverlay";
```

Comment:
update 8/16: this is the second line that is different between `fides-tcf.ts` and `fides.ts`. I chose to use two separate files for `renderOverlay` since that's a `tsx` file, and I thought changing `fides.ts` to `fides.tsx` would cause more downstream changes